### PR TITLE
SLING-11567 add null checks

### PR DESCRIPTION
**Pull request by joerghoh**

### CODE REVIEW

File: src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java
Comment (jsedding) [inline src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java:91]: You should also suppress warnings for "java:S2583" to silence sonar.

**Review by jsedding — APPROVED** [review]
I would suppress the related sonar warnings. Other than that it looks good to me.

File: src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java
Comment (joerghoh) [reply src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java]: I decided the remove the ```@NotNull``` annotations instead, because they are misleading (and actually wrong).

Comment (kwin) [reply src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java]: They are good to enforce correct API usage. IMHO you should just call get with a single argument if you don’t care about the type or a default value!

Comment (joerghoh) [reply src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java]: the API docs does have the ```@NonNull``` annotation and it makes sense here. Here in this implementation it's misleading.

Comment (jsedding) [reply src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java]: My IDE complains when the annotations are present on the interface but not on the implementation. I can see your POV, but in this case I would lean strongly towards `@SuppressWarnings`. IMHO it's exactly a case like this where adding this annotation documents that the "misimplementation" of the interface is deliberate, and thus encourages interested future maintainers to dig through commit messages and JIRAs in order to understand the rationale.

Comment (joerghoh) [reply src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java]: thanks @kwin  and @jsedding  for your suggestions. I added the additional parameter to ```@SuppressWarning``` and re-added the ```@NotNull``` annotation, but left the comment on top.

File: src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java
Comment (kwin) [inline src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java:90]: invalid javadoc, rather `{code type}`

Comment (kwin) [reply src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java:90]: The taglet format is `{@code }`: https://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#code

Comment (joerghoh) [reply src/main/java/org/apache/sling/jcr/resource/internal/JcrValueMap.java:90]: my bad (first save, then commit)

**Comment by joerghoh** [comment]
Want to wait for feedback from @anchela as she has done these changes.

**Review by anchela — APPROVED** [review]
@joerghoh , given the circumstances the patch looks good to me. thanks a lot.

**Pull request merged**
